feat: Cloud Run Documentation (#463)
observIQ/observiq-otel-collector: docs/google-cloud-run.md

## Cloud Run

The collector can be deployed with [Google Cloud Run](https://cloud.google.com/run).

## Image

Push the collector image to your GCR account:
- replace `1.1.0` with your desired version
- replace `myproject` with your gcr project

```bash
docker pull observiq/observiq-otel-collector:1.1.0
docker tag observiq/observiq-otel-collector:1.1.0 gcr.io/myproject/observiq-otel-collector:1.1.0
docker push gcr.io/myproject/observiq-otel-collector:1.1.0
```

> ## Deployment

Author: I've added a disclaimer at the start that the example is Mongo Atlas, but it can be used for other things.

Follow the steps in [Image](./google-cloud-run.md#image) before continuing.

**Secrets**

1. Create secret: cloudrun-mongodb-publickey (mongodb atlas public key)
2. Create secret: cloudrun-mongodb-privatekey (mongodb atlas private key)
3. Create secret: cloudrun-mongodb-config  (collector config.yaml, upload as a file)
4. Give the Cloud Run service account "secret accessor" for each secret (from steps 1-3)

![Secrets](assets/secrets.png)

> ![Sample Config](assets/config.example.yaml)

Reviewer: ```suggestion
[Sample Config](assets/config.example.yaml)
```

## Cloud Run

> This repo servces as a proof of concept for deploying [observiq-otel-collector](https://github.com/observIQ/observiq-otel-collector) to
> [Google Cloud Run](https://cloud.google.com/run). The configuration used in the example is for Mongodb Atlas.

Reviewer 2: ```suggestion
The collector can be deployed with [Google Cloud Run](https://cloud.google.com/run).
```

## Image

Push the collector image to your GCR account:
- replace `1.1.0` with your desired version
- replace `myproject` with your gcr project

```bash
docker pull observiq/observiq-otel-collector:1.1.0
docker tag observiq/observiq-otel-collector:1.1.0 gcr.io/myproject/observiq-otel-collector:1.1.0
docker push gcr.io/myproject/observiq-otel-collector:1.1.0
```

## Deployment

> Follow the steps in [Image](./README.md#image) before continuing.

Reviewer 2: ```suggestion
Follow the steps in [Image](./google-cloud-run.md#image) before continuing.
```
